File: item_scanner.py
inventory = {}

def add_item(name, quantity, expiration_date, slot):
    if name not in inventory:
        inventory[name] = {
            "quantity": 0,
            "expiration_date": "",
            "slot": 0
        }
    inventory[name]["quantity"] += quantity
    inventory[name]["expiration_date"] = expiration_date

def display_inventory():
    print("Current Inventory:")
    for name, item_info in inventory.items():
        total_quantity = item_info["quantity"]
        expiration_date = item_info["expiration_date"]
        print(f"{name}: {total_quantity} items, Expiration Date: {expiration_date}")

while True:
    print("\nStock Item (or type 'quit' to exit):")

    name = input("Enter item name: ").strip()

    if name.lower() == "quit":
        break
    
    try:
        quantity = int(input("Enter quantity: "))
        expiration_date = input("Enter expiration date (YYYY-MM-DD): ")
        slot = int(input("Enter slot: "))
        
        add_item(name, quantity, expiration_date, slot)
        print(f"{quantity} {name} added to inventory in slot {slot}.")
    except ValueError:
        print("Invalid input. Please enter a valid quantity.")

display_inventory()

File: item_scanner.cpp
#include <iostream>
#include <iomanip>
#include <map>

std::map<std::string, std::map<std::string, std::string>> inventory;

void addItem(const std::string& name, int quantity, const std::string& expirationDate, int slot) {
    if (inventory.find(name) == inventory.end()) {
        inventory[name] = {
            {"quantity", "0"},
            {"expirationDate", ""},
            {"slot", "0"}
        };
    }

    inventory[name]["quantity"] = std::to_string(std::stoi(inventory[name]["quantity"]) + quantity);
    inventory[name]["expirationDate"] = expirationDate;
    inventory[name]["slot"] = std::to_string(slot);
}

void displayInventory() {
    std::cout << "Current Inventory:" << std::endl;
    for (const auto& [name, itemInfo] : inventory) {
        int totalQuantity = std::stoi(itemInfo.at("quantity"));
        std::string expirationDate = itemInfo.at("expirationDate");
        std::string slot = itemInfo.at("slot");
        std::cout << name << ": " << totalQuantity << " items, Expiration Date: " << expirationDate
                  << ", Slot: " << slot << std::endl;
    }
}

int main() {
    while (true) {
        std::cout << "\nStock Item (or type 'quit' to exit):" << std::endl;

        std::string name;
        std::cout << "Enter item name: ";
        std::getline(std::cin, name);
        name.erase(name.find_last_not_of(" \n\r\t") + 1);

        if (name == "quit") {
            break;
        }

        try {
            int quantity;
            std::cout << "Enter quantity: ";
            std::cin >> quantity;

            std::string expirationDate;
            std::cout << "Enter expiration date (YYYY-MM-DD): ";
            std::cin >> expirationDate;

            int slot;
            std::cout << "Enter slot: ";
            std::cin >> slot;

            addItem(name, quantity, expirationDate, slot);
            std::cout << quantity << " " << name << " added to inventory in slot " << slot << "." << std::endl;
        } catch (...) {
            std::cout << "Invalid input. Please enter a valid quantity." << std::endl;
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        }
    }

    displayInventory();

    return 0;
}
